File: SuboptimalProcedureLearner/SuboptimalProcedureLearner/SuboptimalProcedureLearner/CommunicationChannel.cs
using System;
using System.Collections.Generic;

using Datastructures;

namespace SuboptimalProcedureLearner.SuboptimalProcedureLearner
{
    class CommunicationChannel
    {
        public void signalExternalOperator(string name, List<Variadic> parameters)
        {
            Console.WriteLine("EXT-OP (" + name + ") parameters [" + convertParametersToString(parameters) + "]");
        }

        private static string convertParametersToString(List<Variadic> parameters)
        {
            int parametersLastIndex;
            int i;
            string result;

            parametersLastIndex = parameters.Count-1;
            
            result = "";

            for( i = 0; i < parameters.Count; i++ )
            {
                result += parameters[i].toString();

                if( i != parametersLastIndex )
                {
                    result += ", ";
                }
            }

            return result;
        }


    }
}

File: src/java/ptrman/agix0/SuboptimalProcedureLearner/CommunicationChannel.java
package ptrman.agix0.SuboptimalProcedureLearner;

import ptrman.Datastructures.Variadic;

import java.util.List;

public class CommunicationChannel {
    public void signalExternalOperator(String name, List<Variadic> parameters) {
        System.out.println("EXT-OP (" + name + ") parameters [" + convertParametersToString(parameters) + "]");
    }

    private static String convertParametersToString(List<Variadic> parameters) {
        int parametersLastIndex = parameters.size() - 1;
        String result = "";
        for( int i = 0;i < parameters.size(); i++ ) {
            result += parameters.get(i).toString();
            if (i != parametersLastIndex) {
                result += ", ";
            }
             
        }
        return result;
    }

}
